### PR TITLE
[new release] argon2 (1.0.2)

### PR DESCRIPTION
**Pull request by Khady**
OCaml bindings to Argon2

- Project page: <a href="https://github.com/Khady/ocaml-argon2">https://github.com/Khady/ocaml-argon2</a>
- Documentation: <a href="https://khady.github.io/ocaml-argon2/">https://khady.github.io/ocaml-argon2/</a>

### CODE REVIEW

File: packages/argon2/argon2.1.0.2/opam [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "ocaml" {>= "4.02.3"}
+  "dune" {>= "2.0"}
+  "dune-configurator" {>= "2.0"}
+  "ctypes" {>= "0.4.1"}
+  "ctypes-foreign"
+  "result"
+  "odoc" {with-doc}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (samoht) [inline packages/argon2/argon2.1.0.2/opam:19]:
```suggestion
  ["dune" "subst"] {dev}
```

**Comment by samoht** [comment]
this seems to be missing a `depext`?

**Comment by MarcCoquand** [comment]
Hi! Any way we can help out to get this merge? It resolves not only the issue with argon2 not working on OS X but also with Nix

**Comment by Khady** [comment]
I’ve added the depexts for the systems I know about

**Comment by mseri** [comment]
There seems to be linking issues though:
```
#=== ERROR while compiling FPauth-strategies.1.0.0 ============================#
# context              2.2.0~alpha4~dev | linux/x86_64 | ocaml-base-compiler.4.14.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.14/.opam-switch/build/FPauth-strategies.1.0.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p FPauth-strategies -j 255
# exit-code            1
# env-file             ~/.opam/log/FPauth-strategies-7-dd1073.env
# output-file          ~/.opam/log/FPauth-strategies-7-dd1073.out
### output ###
# File "test/strategies/dune", line 2, characters 7-11:
# 2 |  (name unit)
#            ^^^^
# (cd _build/default/test/strategies && ./unit.exe)
# Fatal error: exception Dl.DL_error("./unit.exe: undefined symbol: argon2_type2string")
# Raised at Dl._report_dl_error in file "src/ctypes-foreign/dl.ml.unix", line 44, characters 20-44
# Called from Ctypes_foreign_basis.Make.foreign in file "src/ctypes-foreign/ctypes_foreign_basis.ml", line 47, characters 19-47
# Re-raised at Ctypes_foreign_basis.Make.foreign in file "src/ctypes-foreign/ctypes_foreign_basis.ml", line 49, characters 50-59
# Called from Argon2.Kind.argon2_type2string in file "src/argon2.ml", line 18, characters 4-98
```

**Comment by samoht** [comment]
@Khady could you have a look at the linking issue?

```
# Fatal error: exception Dl.DL_error("./unit.exe: undefined symbol: argon2_type2string")
```

Is this expected?

**Comment by samoht** [comment]
I'm merging the PR anyway as it seems to fix user issues (from @MarcCoquand report). But feel free to investigate why one  reverse dependency is broken :-)

**Comment by Khady** [comment]
the linking issue is not expected. It is being tracked in https://github.com/Khady/ocaml-argon2/issues/13

**Comment by hannesm** [comment]
may the linking issue be related / identical to https://discuss.ocaml.org/t/how-to-properly-define-a-library-with-c-bindings/14419/5 ? (I've no clue, and am not sure whether argon uses ctypes)

**Pull request merged**